Sources/Action.swift

```diff
@@ -31,34 +31,12 @@ public final class Action<Input, Output, Error: Swift.Error> {
 	public let errors: Signal<Error, NoError>
 
 	/// Whether the action is currently executing.
-	public var isExecuting: Property<Bool> {
-		return Property(_isExecuting)
-	}
-
-	private let _isExecuting: MutableProperty<Bool> = MutableProperty(false)
+	public let isExecuting: Property<Bool>
 
 	/// Whether the action is currently enabled.
-	public var isEnabled: Property<Bool> {
-		return Property(_isEnabled)
-	}
-
-	private let _isEnabled: MutableProperty<Bool> = MutableProperty(false)
-
-	/// Whether the instantiator of this action wants it to be enabled.
-	private let isUserEnabled: Property<Bool>
+	public let isEnabled: Property<Bool>
 
-	/// This queue is used for read-modify-write operations on the `_executing`
-	/// property.
-	private let executingQueue = DispatchQueue(
-		label: "org.reactivecocoa.ReactiveSwift.Action.executingQueue",
-		attributes: []
-	)
-
-	/// Whether the action should be enabled for the given combination of user
-	/// enabledness and executing status.
-	private static func shouldBeEnabled(userEnabled: Bool, executing: Bool) -> Bool {
-		return userEnabled && !executing
-	}
+	private let _state: MutableProperty<ActionState>
```

Comment:
Our style has been to not prefix private variables with an underscore unless there's a public variable with the same name.